apps/pay/components/Memo/index.tsx

```diff
@@ -12,35 +13,32 @@ interface Props {
 }
 
 const Memo = ({ state, dispatch }: Props) => {
-  const router = useRouter()
-  const { username, amount, sats, unit, memo, display } = router.query
+  const searchParams = useSearchParams()
+  const { username } = useParams()
+  const query = searchParams ? Object.fromEntries(searchParams.entries()) : {}
+
+  const { amount, sats, unit, memo, display } = query
   const [openModal, setOpenModal] = React.useState<boolean>(false)
   const [note, setNote] = React.useState<string>(memo?.toString() || "")
 
   const handleSetMemo = () => {
     if (unit === "SAT" || unit === "CENT") {
-      router.push(
-        {
-          pathname: `${username}`,
-          query: {
-            amount: amount,
-            sats: sats,
-            unit: unit,
-            memo: note,
-            display,
-          },
-        },
-        undefined,
-        { shallow: true },
+      window.history.pushState(
+        {},
+        "",
+        `${username}?${new URLSearchParams({
+          amount: amount,
+          sats: sats,
+          unit: unit,
+          memo: note,
+          display,
+        }).toString()}`,
       )
     } else {
-      router.push(
```

Comment:
shouldn't we use the router.push as it's shown here? 

https://nextjs.org/docs/app/api-reference/functions/use-search-params#updating-searchparams

why using the `window.history.pushState` directly?

this sounds very low level to me, and unclear why we would need it?

Reply:
note: there are multiple places in the code where this comment applies

Reply:
App router doesn't support shallow routing: https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating#shallow-routing.

So, this is equivalent to that now. I am replacing these by modifying other parts so everything works together perfectly.

Currently trying to make this work together as in page router, then will do cleanup.

Reply:
this page is from the `page` router. isn't this the equivalent for the app router? https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#5-soft-navigation

Reply:
> this page is from the `page` router. isn't this the equivalent for the `app` router? https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#5-soft-navigation

Yes, page router had this option and app router don't.